zerver/lib/actions.py

```diff
@@ -1284,6 +1285,93 @@ def do_delete_user(user_profile: UserProfile) -> None:
         )
 
 
+def do_delete_user_preserving_messages(user_profile: UserProfile) -> None:
+    """
+    This is a version of do_delete_user which does not delete message content.
```

Comment:
Maybe add a sentence about the motivation -- you want to delete a user but preserve the integrity of conversations in which they were a significant participant.